examples/evm/its-lock-unlock-fee/FeeToken.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { InterchainTokenStandard } from '@axelar-network/interchain-token-service/contracts/interchain-token/InterchainTokenStandard.sol';
+import { ERC20 } from '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+
+import { Minter } from '@axelar-network/interchain-token-service/contracts/utils/Minter.sol';
+
+/**
+ * @title FeeToken
+ * @notice This contract implements an interchain token which extends InterchainToken functionality and extracts a fee for each transfer.
+ * @dev This contract also inherits Minter and Implementation logic.
+ */
+contract FeeToken is InterchainTokenStandard, ERC20, Minter {
+    uint8 internal immutable decimals_;
+    bytes32 internal tokenId;
+    address internal immutable interchainTokenService_;
+
+    uint256 internal constant UINT256_MAX = 2 ** 256 - 1;
+
+    uint256 public feePercent;
+
+    /**
+     * @notice Constructs the InterchainToken contract.
+     * @dev Makes the implementation act as if it has been setup already to disallow calls to init() (even though that would not achieve anything really).
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimalsValue,
+        uint256 _feePercent,
+        address _interchainTokenServiceAddress
+    ) ERC20(_name, _symbol) {
+        decimals_ = _decimalsValue;
+        interchainTokenService_ = _interchainTokenServiceAddress;
+
+        feePercent = _feePercent;
+
+        _addMinter(msg.sender);
+    }
+
+    function decimals() public view override returns (uint8) {
+        return decimals_;
+    }
+
+    /**
+     * @notice set new fee percent
+     * @param _feePercent new fee percent
+     */
+    function setFeePercent(uint256 _feePercent) external onlyRole(uint8(Roles.MINTER)) {
```

Comment:
why uint256 for feePercent when it has to be < 100?

Reply:
the percentage is large than 100 because the token is 1e18 decimal points. In the deploy script itself I set the fee percentage to 5e16 to be 5% of 1e18.